Check if buckets are outdated in `scoop status`

nimzo6689 opened the pull request:
https://github.com/lukesampson/scoop/issues/3549

Ash258 commented:
Dupliacting code from update is not key.

completely easy, non resource consuming fix could look like this:,

```powershell
Test-UpdateStatus {
    Get-LocalBucket | Foreach-Object {
          $loc = Find-BucketDirectory $_ -Root
          Push-Location $loc
          git_fetch -q origin
          if ((git_status) -like 'is behind') { return $true }
    }
}
```

nimzo6689 commented:
@Ash258 
Thanks for the advice.
I've updated some codes for refactoring.

*In PowerShell, `return` in Foreach-Object behaves like the continue statement. (I missed it.)
So I replaced it with foreach statement for return early.

How about this?

r15ch13 commented on libexec/scoop-status.ps1:
```diff
@@ -10,25 +10,37 @@
+    $is_behind = (git_status) -like '*is behind*' -as [bool]
     Pop-Location
+
+    return !($is_behind)
```
Also renaming to `Test-IsGitRepositoryBehind` would make the function more clear.

nimzo6689 replied:
Done.

nimzo6689 commented on libexec/scoop-status.ps1:
```diff
@@ -10,25 +10,37 @@
-if($needs_update) {
+# check if scoop needs updating
+$needs_update = !(Test-ScoopLatest)
+if ($needs_update) {
```
`$needs_update` is also used in L115.

nimzo6689 commented on libexec/scoop-status.ps1:
```diff
@@ -10,25 +10,37 @@
-    $needs_update = $true
+
+function Test-ScoopLatest {
+    $currentdir = fullpath $(versiondir 'scoop' 'current')
```
Done.
